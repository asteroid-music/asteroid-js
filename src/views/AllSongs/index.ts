//Main component
export { AsteroidSongList } from './AllSongs';


