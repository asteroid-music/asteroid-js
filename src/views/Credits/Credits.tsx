//Basic imports
import React from 'react';
import Typography from '@material-ui/core/Typography';
import Container from '@material-ui/core/Container';

/**
 * <AsteroidCredits> component for main body
 */
function AsteroidCredits(props: {}) {
    return <Container>
        <Typography variant="body1">
            Asteroid built by <a href="https://github.com/dustpancake" target="_blank">
                Fergus Baker
            </a>
            {" and "}
            <a href="https://github.com/JR-Mitchell" target="_blank">
                JR Mitchell
            </a>
            <br/>
            {"GitHub repo at: "}
            <a href="https://github.com/asteroid-music" target="_blank">
                asteroid-music
            </a>
        </Typography>
    </Container>
}


export { AsteroidCredits }
