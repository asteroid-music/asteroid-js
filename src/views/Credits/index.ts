//Main component
export { AsteroidCredits } from './Credits';


