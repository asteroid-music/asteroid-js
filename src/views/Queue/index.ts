//Main component
export { AsteroidQueue } from './Queue';


