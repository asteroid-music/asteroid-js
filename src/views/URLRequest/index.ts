//Main component
export { AsteroidURLRequester } from './URLRequest';


