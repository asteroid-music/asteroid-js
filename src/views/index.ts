export { AsteroidCredits } from './Credits';
export { AsteroidQueue } from './Queue';
export { AsteroidURLRequester } from './URLRequest';
export { AsteroidSongList } from './AllSongs';


