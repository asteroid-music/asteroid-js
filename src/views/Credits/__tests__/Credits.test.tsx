import React from 'react';
import renderer from 'react-test-renderer';
import { AsteroidCredits as Credits } from 'views/Credits';

describe('<Credits />',()=>{
    //Basic rendering with minimal props
    describe('basic rendering',()=>{
        it("renders correctly",()=>{
            expect(renderer
                .create(<Credits/>)
                .toJSON()
            ).toMatchSnapshot();
        });
    });
});