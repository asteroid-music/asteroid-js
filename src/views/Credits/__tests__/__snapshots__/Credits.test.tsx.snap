// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Credits /> basic rendering renders correctly 1`] = `
<Material UI Container>
  <Material UI Typography
    variant="body1"
  >
    Asteroid built by 
    <a
      href="https://github.com/dustpancake"
      target="_blank"
    >
      Fergus Baker
    </a>
     and 
    <a
      href="https://github.com/JR-Mitchell"
      target="_blank"
    >
      JR Mitchell
    </a>
    <br />
    GitHub repo at: 
    <a
      href="https://github.com/asteroid-music"
      target="_blank"
    >
      asteroid-music
    </a>
  </Material UI Typography>
</Material UI Container>
`;
